### PR TITLE
Adding ruleset to some projects - part 01

### PR DESCRIPTION
**Pull request by eddynaka**
Refs #156

- Api
- Collector.AspNet.Win
- Collector.AspNetCore
- Collector.AspNetCore.Tests

### CODE REVIEW

File: test/OpenTelemetry.Collector.AspNetCore.Tests/TestCallbackMiddlewareImpl.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// limitations under the License.
+// </copyright>
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using TestApp.AspNetCore._3._1;
+
+namespace OpenTelemetry.Collector.AspNetCore.Tests
+{
+    public class TestCallbackMiddlewareImpl : CallbackMiddleware.CallbackMiddlewareImpl
```
Comment (bruno-garcia) [inline test/OpenTelemetry.Collector.AspNetCore.Tests/TestCallbackMiddlewareImpl.cs:23]: If I understand, this type was only used on `IncomingRequestsCollectionsIsAccordingToTheSpecTests` and was a inner class. What's the reasoning to lift this to its own file as a public class?

Comment (eddynaka) [reply test/OpenTelemetry.Collector.AspNetCore.Tests/TestCallbackMiddlewareImpl.cs:23]: We have one rule that we cant have a method and a class together. https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md

I saw that if i move to the end of the file, this stops the error. will update! thanks for reviewing!

Comment (bruno-garcia) [reply test/OpenTelemetry.Collector.AspNetCore.Tests/TestCallbackMiddlewareImpl.cs]: Nice. Thanks

File: test/OpenTelemetry.Collector.AspNetCore.Tests/OpenTelemetry.Collector.AspNetCore.Tests.csproj [hunk 2]
```diff
@@ -4,6 +4,10 @@
     <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <CodeAnalysisRuleSet>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), 'OpenTelemetry.sln'))/build/OpenTelemetry.prod.ruleset</CodeAnalysisRuleSet>
```
Comment (SergeyKanzhelev) [inline test/OpenTelemetry.Collector.AspNetCore.Tests/OpenTelemetry.Collector.AspNetCore.Tests.csproj:8]: can you please revert it to test
```suggestion
    <CodeAnalysisRuleSet>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), 'OpenTelemetry.sln'))/build/OpenTelemetry.test.ruleset</CodeAnalysisRuleSet>
```

**Pull request closed without merging**